pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1149,8 +1151,21 @@ cdef class _Timedelta(timedelta):
         if isinstance(other, _Timedelta):
             ots = other
         elif is_any_td_scalar(other):
-            ots = Timedelta(other)
-            # TODO: watch out for overflows
+            try:
+                ots = Timedelta(other)
+            except OutOfBoundsTimedelta as err:
+                # GH#49021 pytimedelta.max overflows
+                if not PyDelta_Check(other):
+                    # TODO: handle this case
+                    raise
+                if op == Py_EQ:
+                    return False
+                elif op == Py_NE:
+                    return True
+                elif op == Py_LE or op == Py_LT:
+                    return self.days <= other.days
```

Comment:
1. Shouldn't we use `<=` or `PY_LE` and `<` for `Py_LT`?
2. What about the other components?

Reply:
hmm i wrote this with self.unit == "ns" in mind, in which case if it overflows we really just need to know which direction it overflowed.  but with "s"/"ms" this could be subtly wrong.  will update w/ tests.